test/integration/guest_can_change_quanity_in_cart_test.rb

```diff
@@ -0,0 +1,43 @@
+require 'test_helper'
+
+class GuestCanChangeQuanityInCartTest < ActionDispatch::IntegrationTest
+  test "guest visits cart and adds more items" do
+    create_and_add_item_to_cart
+    visit cart_path
+    fill_in "Quantity", with: "4"
+    click_button "Save changes"
+    # save_and_open_page
+    assert_equal '/cart', current_path
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.